src/ScottPlot/Plot/Plot.Axis.cs

```diff
@@ -362,13 +362,13 @@ public AxisLimits GetDataLimits(int xAxisIndex = 0, int yAxisIndex = 0)
 
                 AxisLimits limits = plottable.GetAxisLimits();
 
-                if (xAxisMatch)
+                if (xAxisMatch && !double.IsNaN(limits.XMin) && !double.IsNaN(limits.XMax))
```

Comment:
I think it would be more correct to check `XMin` and `XMax` separately:

```cs
if(xAxisMatch) {
    xMin = !double.IsNaN(limits.XMin) ? Math.Min(xMin, limits.XMin) : XMin;
    xMax = !double.IsNaN(limits.XMax) ? Math.Max(xMax, limits.XMin) : XMax;
}
```

I know it's tedious, but spans and lines can mix finite and infinite bounds.

Reply:
Oof, what a minefield!

Actually I'm pretty sure just checking for `double.NaN` like @EFeru did here will be okay... the only way an axis would have a `double.PositiveInfinity` or `double.NegativeInfinity` limit is if the user sets its position to that intentionally, in which case they get what they deserve 😈 

This is where axis limits are returned. It looks like `double.NaN` is returned for the non-positional axes:
https://github.com/ScottPlot/ScottPlot/blob/d9cd05eaafc9fbeff19870dde7548fc9a9ce8a83/src/ScottPlot/Plottable/AxisSpan.cs#L90-L99

Reply:
> I think it would be more correct to check XMin and XMax separately

I think I misunderstood this comment originally 😅 

Yes, I agree! Something like this:
```cs
if (!double.IsNaN(limits.XMin))
    xMin = Math.Min(xMin, limits.XMin); 

if (!double.IsNaN(limits.XMax))
    xMax = Math.Max(xMax, limits.XMax); 

/* etc... */
```